syntax = "proto3";
package minecpp.proto.event.serverbound.v1;

import "google/protobuf/any.proto";
import "minecpp/proto/common/v1/common.proto";
import "minecpp/proto/player/v1/id.proto";
import "minecpp/proto/entity/v1/entity.proto";

message Event {
  google.protobuf.Any payload = 1;
  player.v1.PlayerId player_id = 2;
}

message AcceptPlayer {
  string name = 1;
  uint32 challenge_id = 2;
  uint32 front_id = 3;
}

message SetPlayerPosition {
  entity.v1.Position position = 1;
}

message SetPlayerRotation {
  entity.v1.Rotation rotation = 1;
}

message ChatMessage {
  string message = 1;
}

message RemovePlayer {
}

message DestroyBlock {
  common.v1.BlockPosition block_position = 1;
}

message UpdatePing {
  int32 ping = 1;
}

message AnimateHand {
  int32 hand = 1;
}

message LoadInitialChunks {
}